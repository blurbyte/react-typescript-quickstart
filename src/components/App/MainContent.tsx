import styled from '@styles/styledComponents';
import media from '@styles/media';

import Content from '../Content';

const Article = Content.withComponent('article');

const MainContent = styled(Article)`
  display: grid;
  grid-template-columns: repeat(2, 1fr) minmax(30rem, 1fr);
  grid-gap: 0 3rem;
  padding-bottom: 6rem;
  padding-top: 9rem;

  ${media.tablet`
    grid-template-columns: repeat(3, 1fr);
    padding-top: 6rem;
  `};
`;

export default MainContent;
