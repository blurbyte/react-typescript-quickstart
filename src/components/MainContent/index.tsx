import styled from '@styles/styledComponents';
import Content from '../Content';

const Article = Content.withComponent('article');

const MainContent = styled(Article)`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 2rem;
`;

export default MainContent;
