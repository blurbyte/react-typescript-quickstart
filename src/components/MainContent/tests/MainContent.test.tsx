import * as React from 'react';
import { shallow } from 'enzyme';

import MainContent from '../index';

describe('MainContent', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<MainContent />);
    expect(wrapper).toMatchSnapshot();
  });
});
