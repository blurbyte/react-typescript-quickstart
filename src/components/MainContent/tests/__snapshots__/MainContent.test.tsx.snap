// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  grid-gap: 2rem;
}

<styled.article
  className="c0"
/>
`;
